src/callAPIMethod.js

```diff
@@ -33,17 +33,20 @@ export default function callAPI(
     if (body) accumulatedFetchOptions.body = body;
 
     return fetch(url, accumulatedFetchOptions)
-        .then( response => {
-            const responseBodyPromise = response.body ? response[method]() : Promise.resolve(response.body);
-
-            if (response.ok) {
-                return responseBodyPromise;
-            }
-
-            return responseBodyPromise
-                .then( responseBody => {
-                    throw new APIError(response.status, response.statusText, responseBody);
-                });
-        })
+        .then( response => response[method]()
+                .catch( () => {
+                    if (!response.ok) {
```

Comment:
Let's keep the same [codestyle](https://github.com/auru/unity-api/blob/master/src/callAPIMethod.js#L32) everywhere